packages/api/src/routes/medical/internal-patient.ts

```diff
@@ -1,47 +1,36 @@
 import Router from "express-promise-router";
 import status from "http-status";
 import stringify from "json-stringify-safe";
 import { z } from "zod";
-import { getFacilities, getFacilityOrFail } from "../../command/medical/facility/get-facility";
+import { getFacilityOrFail } from "../../command/medical/facility/get-facility";
 import { deletePatient } from "../../command/medical/patient/delete-patient";
-import {
-  getPatientIds,
-  getPatientOrFail,
-  getPatients,
-} from "../../command/medical/patient/get-patient";
-import { PatientUpdateCmd, updatePatient } from "../../command/medical/patient/update-patient";
-import { Patient } from "../../domain/medical/patient";
+import { getPatientIds, getPatientOrFail } from "../../command/medical/patient/get-patient";
+import { PatientUpdaterCommonWell } from "../../external/commonwell/patient-updater-commonwell";
 import { getFacilityIdOrFail } from "../../domain/medical/patient-facility";
-import { processAsyncError } from "../../errors";
 import BadRequestError from "../../errors/bad-request";
 import { MedicalDataSource } from "../../external";
+import { getCxsWithEnhancedCoverageFeatureFlagValue } from "../../external/aws/appConfig";
 import cwCommands from "../../external/commonwell";
 import { findDuplicatedPersons } from "../../external/commonwell/admin/find-patient-duplicates";
 import { patchDuplicatedPersonsForPatient } from "../../external/commonwell/admin/patch-patient-duplicates";
 import { recreatePatientsAtCW } from "../../external/commonwell/admin/recreate-patients-at-hies";
+import { checkStaleEnhancedCoverage } from "../../external/commonwell/cq-bridge/coverage-enhancement-check-stale";
+import { completeEnhancedCoverage } from "../../external/commonwell/cq-bridge/coverage-enhancement-complete";
+import { initEnhancedCoverage } from "../../external/commonwell/cq-bridge/coverage-enhancement-init";
+import { cqLinkStatus } from "../../external/commonwell/patient-shared";
 import { getETag } from "../../shared/http";
 import { errorToString } from "../../shared/log";
+import { capture } from "../../shared/notifications";
 import { stringToBoolean } from "../../shared/types";
-import { getUUIDFrom } from "../schemas/uuid";
+import { stringIntegerSchema } from "../schemas/shared";
+import { getUUIDFrom, uuiSchema } from "../schemas/uuid";
 import { asyncHandler, getFrom, getFromParamsOrFail } from "../util";
-import { dtoFromCW, PatientLinksDTO } from "./dtos/linkDTO";
+import { PatientLinksDTO, dtoFromCW } from "./dtos/linkDTO";
 import { linkCreateSchema } from "./schemas/link";
 
 const router = Router();
 
-async function updateInFHIRAndCW(
-  patientUpdate: PatientUpdateCmd,
-  facilityId: string
-): Promise<void> {
-  const updatedPatient = await updatePatient(patientUpdate);
-  // Intentionally asynchronous - it takes too long to perform
-  cwCommands.patient
-    .update(updatedPatient, facilityId)
-    .catch(processAsyncError(`cw.patient.update`));
-}
```

Comment:
This is important: removed the updatePatient from this logic (moved to [here](https://github.com/metriport/metriport/pull/1163/files#diff-c1495f1ae800a5da072464407d2f89f3186c7f509ff379a680cbc21d25d12078R28)), only updating CW now. Couldn't see any reason for it and I think it could be the reason for the N+1 notifs we're getting on Slack/Sentry.

Reply:
Yup agreed, IIRC it was this way originally for speed - ie just to re-use the existing patient update logic, but in bulk, no reason to update the DB/FHIR if demo isn't changing